src/index.js

```diff
@@ -36,6 +36,15 @@
  *      }
  *    }
  *
+ * Some ES providers like Bonsai limit the number of concurrent reads/writes, and respond with a 429 error over
```

Comment:
Okay, it seems "nice" to expose this as a configuration option, but I wonder if you aren't just expanding the attack surface. If you're fiddling with retry settings, it seems like you're in trouble. The retry code should "just work."

Reply:
Yeah, I just exposed these to avoid us having to re-release if we find these need to be experimented with to find a good balance - ultimately these defaults should be updated to what we find works, if these first guesses aren't good.